docs/advanced/porting_guide.md

```diff
@@ -36,6 +36,52 @@ Add the target description to ```mbed-os\targets\targets.json```:
 
 See the [mbed Target Documentation](mbed_targets.md) for more details on what this definition means.
 
+### Coding requirements
+
+All code submitted as part of a port must compile with GCC ARM Embedded, ARM Compiler 5 and IAR EWARM. Flags that port code must compile with for each compiler are listed below. The profiles used to build mbed OS are subsets of each of these flags with exactly one optimization setting.
+
+#### GCC ARM Embedded
+
+Code in a port must be compatible with the `softfp` float ABI and compile without warnings or errors using GCC ARM Embedded with the following flags.
```

Comment:
This says that the errors use GCC ARM Embedded and flags. Is that correct? If not, please change to "Code in a port that uses GCC ARM Embedded must compile without warnings or errors and must use the following flags" or something similar.